web-admin/src/features/projects/github/GithubConnectionUpdater.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  createAdminServiceConnectProjectToGithub,
+  getAdminServiceGetGithubUserStatusQueryKey,
+  getAdminServiceGetProjectQueryKey,
+  type ListGithubUserReposResponseRepo,
+} from "@rilldata/web-admin/client";
+import { extractGithubConnectError } from "@rilldata/web-admin/features/projects/github/github-errors";
+import { queryClient } from "@rilldata/web-common/lib/svelte-query/globalQueryClient";
+import { behaviourEvent } from "@rilldata/web-common/metrics/initMetrics";
+import { BehaviourEventAction } from "@rilldata/web-common/metrics/service/BehaviourEventTypes";
+import { invalidateRuntimeQueries } from "@rilldata/web-common/runtime-client/invalidation";
+import { get, writable } from "svelte/store";
+
+export class GithubConnectionUpdater {
+  public readonly showOverwriteConfirmation = writable(false);
+  public readonly connectToGithubMutation =
+    createAdminServiceConnectProjectToGithub();
+
+  public readonly githubUrl = writable("");
+  public readonly subpath = writable("");
+  public readonly branch = writable("");
+
+  private isConnected: boolean;
+  private defaultBranch = "";
+
+  public constructor() {}
+
+  public init(githubUrl: string, subpath: string, branch: string) {
+    this.githubUrl.set(githubUrl);
+    this.subpath.set(subpath);
+    this.branch.set(branch);
+    this.isConnected = !!githubUrl;
+  }
```

Comment:
It'd be more idiomatic to implement the constructor, rather than create a separate `init` function

Reply:
Ya that was ideal earlier. But it is simpler to do this than create a new instance everytime.